app/models/concerns/has_coordinates.rb

```diff
@@ -2,33 +2,34 @@ module HasCoordinates
   extend ActiveSupport::Concern
 
   included do
-    geocoded_by :full_address
-    after_validation :geocode, if: ->(obj){ (obj.address.present? || obj.city.present?) and obj.full_address_changed? }
-    after_validation :location_found?
+    after_validation :geocode, if: :geocodable?
```

Comment:
replace `:geocodable?` with `:full_address_changed?` and remove `def geocodable?` since it is now identical.

And it is more explicit, becuse one can see right here when geocodable should or should not run.

Reply:
100% agree, I should have noticed it, wasnt that hard :D